### DISCUSSION TITLE
Erro em float.Parse(Console.ReadLine()); (Calculadora)

### DISCUSSION THREAD

**Discussion by CayoRamos** [balta-io/2801#109] [Perguntas e Respostas] [answered]
Boa Tarde. Em todas as funções que eu tento converter string para float usando o Parse me da o seguinte aviso:
Fica sublinhado embaixo do Console.Readline() ->Possível argumento de referência nula para o parâmetro 's' em 'float float.Parse(string s)'.
Já mudei para Convert.ToSingle e o código não funciona.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @CayoRamos, este aviso ocorre pois diferente de um `float` uma `string` pode ser `nula` (Sem necessidade dos Nullables).
Explico isto na aula de reference e value types.

Desta forma, o compilador te avisa sobre isto.

**Reply by CayoRamos** [reply]
Entendi. Mas o código ele não funciona se aviso estiver ali. Ele fica tentando rodar e da erro
